CONTRIBUTING.adoc

```diff
@@ -131,3 +131,23 @@ Some rules and tips for the python and documentation in this project.
   - Use named parameters, not positional arguments
   - Use the verbs `create`, `show`, `update`, and `remove` for underlying API
       CRUD operations
+
+
+Releasing a Version
+-------------------
+
+If you are a CLI maintainer, you need to be able to do a release.
+
+Setup:
+
+  - Make sure you have a pypi account
+  - Setup your credentials for twine (the pypi upload tool) https://github.com/pypa/twine[twine docs for detail]
+  - Make sure you have a gpg key setup for use with git
+
+Do it:
+
+  - `rm -r dist/` to make sure there are no old or weird source dists lying around
+  - `make upload` to build new source dists in `dist/` and upload via twine
+  - `git tag -s $VERSION` and write `v${VERSION}` in the tag. e.g. `v1.7.0`
```

Comment:
Would there be any appetite to just incorporate these steps into the `Makefile`?

Reply:
Some of it, at least.
Cleaning `dist/` for sure -- I'll even go so far as to do that in this PR.

`VER=$(python -c ...) git tag -s "$VER" -m "v${VER}"`? I'm less sure.

And I don't think the makefile can safely push the tag because we don't all necessarily name the repo the same thing -- `origin` vs `upstream`, etc.

Reply:
> Cleaning `dist/` for sure -- I'll even go so far as to do that in this PR.

Yes, the cleaning for sure -- if having a (partially- or fully-) clean working tree is something you want to happen before people upload for a release, by all means make it automatic.

> `VER=$(python -c ...) git tag -s "$VER" -m "v${VER}"?` I'm less sure.

Yes, I agree this is less fun, but something like `<globus_cli/version.py grep '^__version__' | grep -oe '[0-9.]*'` could also give it to you, though, if you wanted to avoid starting a Python interpreter for some reason.

> And I don't think the makefile can safely push the tag because we don't all necessarily name the repo the same thing

Yes, absolutely, and I think just having an `echo` that just says "please push the new v$(VERSION) tag to the appropriate repository" will keep people from forgetting.